src/connector/src/source/mqtt/enumerator/mod.rs

```diff
@@ -45,95 +30,56 @@ impl SplitEnumerator for MqttSplitEnumerator {
 
     async fn new(
         properties: Self::Properties,
-        context: SourceEnumeratorContextRef,
+        _context: SourceEnumeratorContextRef,
     ) -> ConnectorResult<MqttSplitEnumerator> {
-        let (client, mut eventloop) = properties.common.build_client(context.info.source_id, 0)?;
-
-        let topic = properties.topic.clone();
-        let mut topics = HashSet::new();
-        if !topic.contains('#') && !topic.contains('+') {
-            topics.insert(topic.clone());
-        }
-
-        client
-            .subscribe(topic.clone(), rumqttc::v5::mqttbytes::QoS::AtMostOnce)
-            .await?;
-
-        let topics = Arc::new(RwLock::new(topics));
-
-        let connected = Arc::new(AtomicBool::new(false));
-        let connected_clone = connected.clone();
-
-        let stopped = Arc::new(AtomicBool::new(false));
-        let stopped_clone = stopped.clone();
-
-        let topics_clone = topics.clone();
-        tokio::spawn(async move {
-            while !stopped_clone.load(std::sync::atomic::Ordering::Relaxed) {
-                match eventloop.poll().await {
-                    Ok(Event::Outgoing(Outgoing::Subscribe(_))) => {
-                        connected_clone.store(true, std::sync::atomic::Ordering::Relaxed);
-                    }
-                    Ok(Event::Incoming(Incoming::Publish(p))) => {
-                        let topic = String::from_utf8_lossy(&p.topic).to_string();
-                        let exist = {
-                            let topics = topics_clone.read().await;
-                            topics.contains(&topic)
-                        };
-
-                        if !exist {
-                            let mut topics = topics_clone.write().await;
-                            topics.insert(topic);
-                        }
-                    }
-                    Ok(_)
-                    | Err(ConnectionError::Timeout(_))
-                    | Err(ConnectionError::RequestsDone) => {}
-                    Err(err) => {
-                        tracing::error!(
-                            "Failed to fetch splits to topic {}: {}",
-                            topic,
-                            err.as_report(),
-                        );
-                        tokio::time::sleep(std::time::Duration::from_millis(500)).await
-                    }
-                }
-            }
-        });
-
         Ok(Self {
-            client,
-            topics,
             topic: properties.topic,
-            connected,
-            stopped,
         })
     }
 
     async fn list_splits(&mut self) -> ConnectorResult<Vec<MqttSplit>> {
-        if !self.connected.load(std::sync::atomic::Ordering::Relaxed) {
```

Comment:
we also utilize list_splits func as connectivity test
if setting to a fixed value, the create source statement always succeeds, whether we can connect to the broker or not.